packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -115,18 +115,36 @@ function validateLightDomTemplate(template: Template, vm: VM) {
     }
 }
 
+enum FragmentCache {
+    HAS_TOKEN = 1, // 2^0
+    HAS_SCOPED_STYLE = 2, // 2^1
+    SHADOW_MODE_SYNTHETIC = 4, // 2^2
+}
+
 export function parseFragment(strings: string[], ...keys: number[]): () => Element {
+    const cache = Object.create(null);
+
     return function (): Element {
         const {
             context: { hasScopedStyles, stylesheetToken },
             shadowMode,
         } = getVMBeingRendered()!;
         const hasStyleToken = !isUndefined(stylesheetToken);
+        const isSyntheticShadow = shadowMode === ShadowMode.Synthetic;
+
+        const cacheKey =
+            (hasStyleToken ? FragmentCache.HAS_TOKEN : 0) | // @todo: can this change?
```

Comment:
can the token change for a given template (set/not set) or the token itself changes (ex: `x-component_component` -> `x-foo_component`?

Reply:
No. The token will only either be `undefined` or a string. If it's a string, then it will only be that one string.

Of course developers can monkey-patch it at runtime, in which case all bets are off, but hopefully we can clamp down on this. 😛  #2783

Reply:
🙈 . based on this, we can remove this as a key.

Reply:
edit: while using the same key (`0`) when there's no token. if there's no token, it is going to be the same output always.